testing/benchmark_hooks.py

```
@@ -0,0 +1,142 @@
+    except subprocess.CalledProcessError:
+        # Still record time even if hook fails
+        pass
+    end = time.perf_counter()
+    return end - start
+
+
+def safe_git_restore(files):
+    # Only restore files tracked by git
+    tracked = []
+    for f in files:
+        result = subprocess.run(
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)